src/ol/renderer/canvas/tilelayer.js

```diff
@@ -225,15 +227,25 @@ ol.renderer.canvas.TileLayer.prototype.prepareFrame = function(frameState, layer
         canvas.width = width;
         canvas.height = height;
       } else {
-        context.clearRect(0, 0, width, height);
+        if (this.renderedExtent_ && !ol.extent.equals(imageExtent, this.renderedExtent_)) {
+          context.clearRect(0, 0, width, height);
+        }
         oversampling = this.oversampling_;
       }
     }
 
     this.renderedTiles.length = 0;
     /** @type {Array.<number>} */
     var zs = Object.keys(tilesToDrawByZ).map(Number);
-    zs.sort(ol.array.numberSafeCompareFunction);
+    zs.sort(function(a, b) {
+      if (a === z) {
+        return 1;
+      } else if (b === z) {
+        return -1;
+      } else {
+        return a > b ? 1 : a < b ? -1 : 0;
```

Comment:
Should we use `ol.array.numberSafeCompareFunction` here?

Reply:
My personal opinion is that it is better to have some copied code in cases like this.  I would use `ol.array.numberSafeCompareFunction()` if 1) it was always going to be inlined (this is probably only going to be the case with Closure Compiler), or 2) it involved something more than this trivial expression.

Reply:
OK